dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,395 @@
+
+{|
+!Transaction!!Script!!Description
+|-
+|TxIn[0..N]||Default UTXO scripts||Normal UTXO scripts as you would find on normal transactions.
+|-
+|TxOut[0]||[0]OP_TADD||Value shall be what is credited to the treasury.
+|-
+|TxOut[1]||[0]OP_SSTXCHANGE||Optional if there is no change.
+|}
+
+Additional consensus enforced characteristics:
+* A TAdd transaction shall not have a zero value for change.
+* There shall be no more than twenty TAdd transactions per block.
+
+===Treasury Expenditure Voting===
+
+Voting on valid TSpend transactions is done by piggy-backing on consensus
+voting transactions. The treasury votes are optionally appended to the
+consensus vote.
+
+A treasury vote is identified by the following characteristics:
+* The consensus vote format shall remain identical to the pre decentralized treasury agenda activation save for one optional output.
+* When treasury spend votes are appended the version of the transaction shall be 3.
+* The optional treasury spend vote output shall be the last output of the transaction.
+* The optional treasury spend votes output shall be a valid null script.
+* The null script shall be an <code>OP_RETURN</code> followed by a data push and the first two bytes of the data push shall be 'TV' (or in hex 0x54,0x56).
+* The remainder of the data length shall be 0 modulo of 32+1 (TSpend transaction hash + vote)
+* A TSpend transaction hash shall only occur once.
+* The vote shall be either <code>Yes = 0x01</code> or <code>No = 0x02</code> and all other values shall be rejected. This means the entire consensus vote shall fail.
+
+Since the format of the transaction remained the same only the optional data is
+described in the following tables. Note that there are two valid formats based
+on the number of treasury spend votes that are occurring simultaneously.
+
+Format 1
+{|
+!Transaction!!Script!!Description
+|-
+|TxOut[N-1]||[0]<code>OP_RETURN</code><br>[1]<code>OP_DATA_XX</code><br>[2..3]<code>'TV'</code><br>[4..]<code>treasury vote...</code>||Up to two (2) treasury votes can be encoded using this format.
+|}
+
+Format 2
+{|
+!Transaction!!Script!!Description
+|-
+|TxOut[N-1]||[0]<code>OP_RETURN</code><br>[1]<code>OP_PUSHDATA1</code><br>[2]data length<br>[5...]<code>treasury vote...</code>||Up to seven (7) treasury votes can be encoded.
+|}
+
+Treasury vote
+{|
+!Transaction!!Script!!Description
+|-
+|TxOut[N-1]||[M] TSpend transaction hash<br>[M+32] byte vote bits||Yes vote = 0x01<br>No vote = 0x02<br>All other values shall be rejected rendering the overall transaction invalid.
+|}
+
+
+===Debit treasury===
+
+The TSpend transactions are rather complex and some additional definitions need
+to be made prior to explaining the format.
+
+Required chain configuration parameters:
+* Treasury Vote Interval, or TVI, determines in which blocks a TSpend may occur. A block is considered on a TVI when the block height modulo TVI is equal to zero.
```

Comment:
Does it mean TSpends have a chance to be mined every TVI blocks?

Reply:
Yes.